woocommerce/woocommerce
# Pass order id to manage_{screen_id}_custom_column action

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




Pass the order id as second parameter instead of WC_Order object in HPO action `manage_{screen_id}_custom_column`.
Closes #51750.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add the following hooks to your functions.php
```php
function add_custom_column( $columns ) {
	$columns['my_custom_column'] = 'My custom column';
	return $columns;
}
function display_custom_column( $column_name, $post_id ) {
	if ( 'my_custom_column' !== $column_name && ! is_int( $post_id ) ) {
		return;
	}

	echo 'I am an integer baby ' . $post_id;
}
add_filter( 'manage_woocommerce_page_wc-orders_columns', 'add_custom_column' );
add_action( 'manage_woocommerce_page_wc-orders_custom_column', 'display_custom_column', 10, 2 );
```
2. Got to WooCommerce > Orders.
3. You should see the output 



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
